adodbapi/apibase.py

```diff
@@ -461,20 +465,25 @@ def convert_to_python(variant, func):  # convert DB value into Python value
     return func(variant)  # call the appropriate conversion function
 
 
-class MultiMap(dict):  # builds a dictionary from {(sequence,of,keys) : function}
-    """A dictionary of ado.type : function -- but you can set multiple items by passing a sequence of keys"""
+class MultiMap(Dict[int, Callable[[object], object]]):

```

Comment:
The typing issue here was that `dict` leads to partially `Any` type. Since this is a very small and simple class, I completely typed it.